<a href="https://colab.research.google.com/github/sreejakatta14/machine_learning/blob/main/ml_labtest.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Accuracy using Random Forest Classifier**

In [1]:
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
from sklearn.metrics import accuracy_score, classification_report

In [2]:
iris = load_iris()
X = iris.data  # Independent variables
y = iris.target  # Target variable


In [3]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, random_state=32)


In [4]:
rf_classifier = RandomForestClassifier()


In [5]:
rf_classifier.fit(X_train, y_train)


RandomForestClassifier()

In [6]:
y_pred = rf_classifier.predict(X_test)


In [8]:
accuracy = accuracy_score(y_test, y_pred)
print("Accuracy:", accuracy)


Accuracy: 0.9333333333333333


In [9]:
classification_rep = classification_report(y_test, y_pred)
print("\nClassification Report:\n", classification_rep)


Classification Report:
               precision    recall  f1-score   support

           0       1.00      1.00      1.00        30
           1       0.81      1.00      0.89        21
           2       1.00      0.79      0.88        24

    accuracy                           0.93        75
   macro avg       0.94      0.93      0.93        75
weighted avg       0.95      0.93      0.93        75



**Accuracy using Adaboost Classifier**


In [35]:
# Importing necessary libraries
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.ensemble import AdaBoostClassifier
from sklearn.metrics import accuracy_score
from sklearn.metrics import accuracy_score, classification_report

In [30]:
# Importing the iris dataset
iris = load_iris()
X = iris.data  # Independent variables
y = iris.target  # Target variable

In [31]:
# Splitting the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, random_state=32)

In [32]:
adaboost_clf = AdaBoostClassifier(n_estimators=30, learning_rate=1.0, random_state=42)  # You can adjust n_estimators and learning_rate as per your requirement
adaboost_clf.fit(X_train, y_train)

AdaBoostClassifier(n_estimators=30, random_state=42)

In [33]:
# Making predictions on the test set
y_pred = adaboost_clf.predict(X_test)

In [34]:
# Calculating accuracy
accuracy = accuracy_score(y_test, y_pred)
print("Accuracy:", accuracy)

Accuracy: 0.8933333333333333


In [36]:
classification_rep = classification_report(y_test, y_pred)
print("\nClassification Report:\n", classification_rep)


Classification Report:
               precision    recall  f1-score   support

           0       1.00      1.00      1.00        30
           1       0.72      1.00      0.84        21
           2       1.00      0.67      0.80        24

    accuracy                           0.89        75
   macro avg       0.91      0.89      0.88        75
weighted avg       0.92      0.89      0.89        75



precision : not to label an instance positive that is actually negative

recall : ability to find all the positive instances

f1-score : 2*(Recall * Precision) / (Recall + Precision)




The accuracy of Random Forest is 93.3%

The accuracy of AdaBoost is 89.33%